lms/views/exceptions.py

```diff
@@ -27,12 +27,23 @@ def notfound(self):
 
     @forbidden_view_config()
     def forbidden(self):
-        if validation_error := getattr(self.exception.result, "validation_error", None):
-            self.request.override_renderer = (
-                "lms:templates/validation_error.html.jinja2"
-            )
-            self.request.response.status_int = 403
-            return {"error": validation_error}
+        if exception := getattr(self.exception.result, "exception", None):
+            # If we have an exception attached to the Denied object
+            # we'll treat it the same way the corresponding exception view.
+            try:
+                self.exception = exception
+                raise exception
```

Comment:
Raising and using except to match seems better than using `isinstance` or similar.

Reply:
This is to hook into the exceptions views. Exceptions views don't handle exceptions in the permissions layer so by wrapping the exception in a "Denied" and simulating the right exception view we can have the same behavior.